paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/FlinkSinkBuilder.java

```diff
@@ -119,8 +133,86 @@ public FlinkSinkBuilder inputBounded(boolean bounded) {
         return this;
     }
 
+    /** Set the table sort info. */
+    public FlinkSinkBuilder setTableSortInfo(
+            String sortColumnsString,
+            String sortStrategy,
+            boolean sortInCluster,
+            int sampleFactor) {
+        // 1. The table sort will be ignored if the sort columns are not specified.
+        if (sortColumnsString == null || sortColumnsString.isEmpty()) {
+            return this;
+        }
+        // 2. Check the table type.
+        checkState(
+                table.bucketMode().equals(BUCKET_UNAWARE),
+                "Clustering only supports bucket unaware table without primary keys.");
+        // 3. Check the sort columns.
+        List<String> sortColumns = Arrays.asList(sortColumnsString.split(","));
+        List<String> fieldNames = table.schema().fieldNames();
+        checkState(
+                new HashSet<>(fieldNames).containsAll(new HashSet<>(sortColumns)),
+                String.format(
+                        "Field names %s should contains all clustering column names %s.",
+                        fieldNames, sortColumns));
+        // 4. Check the execution mode.
+        checkState(input != null, "The input stream should be specified earlier.");
+        if (boundedInput == null) {
+            boundedInput = !FlinkSink.isStreaming(input);
+        }
+        checkState(boundedInput, "The clustering should be executed under batch mode.");
```

Comment:
Here we should discuss semantics, maybe we can just not sort in streaming mode?

Reply:
1. +1 for not failing the job. We can enable the feature only when all the conditions are matched. Otherwise, if the `by-columns` are configured, we can print a warning log about it.

2. We probably should also mention the limitations (batch mode, table type, etc.) in the configuration description.

Reply:
@JingsongLi 
Good point, ignoring clustering in stream mode is a sensible design. This avoids the need for users to manually adjust the table configuration under streaming mode.

@xintongsong  
1. I've removed the check and added releated warning log.
2. I've added the limitations of table type and batch mode in configuration description. I've also added a commit to introduce the clustering feature in paimon docs.